src/pages/teamCreation/teamCreation.tsx

```diff
@@ -1,147 +1,169 @@
+  const [newTeamPassword, setNewTeamPassword] = useState('');
+
+  const history = useHistory(); // for routing
+
+  const adData = useContext(AdminContext); // admin context
+
+  // create a new team
+  const createTeam = async () => {
+    const userRef = doc(FirestoreDB, 'users', auth.currentUser.email as string);
+    const userSnap = await getDoc(userRef);
+    const userData = userSnap.data();
+    if (newTeamName === '') {
+      alert('Team name cannot be an empty string!');
+      return;
+    } else {
+      const dbRef = doc(FirestoreDB, 'teams', newTeamName);
+      const dbSnap = await getDoc(dbRef);
+      if (dbSnap.exists()) {
+        alert(`${newTeamName} already exists!`);
+        return;
+      }
+    }
+    if (newTeamStatus == 0 && newTeamPassword !== '') {
+      alert('You cannot create a password for a public team!');
+      return;
+    }
+    if (newTeamStatus == 1 && newTeamPassword === '') {
+      alert('You must create a password for a private team!');
+      return;
+    }
+    const currentDate: Date = new Date();
+    const teamCreationDeadline: Date = new Date(adData.teamDate);
+    if (currentDate > teamCreationDeadline) {
+      alert(
+        `The team creation deadline is: ${teamCreationDeadline}. You cannot create a team now.`
+      );
+      return;
+    }
+    setDoc(doc(FirestoreDB, 'teams', newTeamName), {
+      name: newTeamName,
+      avg_steps: userData.totalStep,
+      leader: userData.email,
+      members: [userData.email],
+      status: newTeamStatus,
+      password: newTeamPassword,
+      profile_pic:
+        'https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460__340.png',
+      totalStep: userData.totalStep,
+      channel_id: '' // TODO: create discord channel
+    })
+      .then(async () => {
+        console.log('Document written successfully');
+        alert('Your team has been created!');
+        await updateCurrentUser();
+        history.push('/app/team');
+      })
+      .catch((error: unknown) => {
+        console.error('Error writing document: ', error);
+      });
+  };
+
+  // update the current user's data
+  const updateCurrentUser = async () => {
+    const currentUserRef = doc(
+      FirestoreDB,
+      'users',
+      auth.currentUser.email as string
+    );
+    await updateDoc(currentUserRef, {
+      team: newTeamName,
+      team_leader: true
+    });
+  };
+
+  return (
+    <IonPage>
+      <IonHeader>
+        <NavBar>
+          <IonTitle>Team Creation</IonTitle>
+        </NavBar>
+      </IonHeader>
+      <IonContent fullscreen className="team-create">
+        <IonHeader collapse="condense">
+          <IonToolbar>
+            <IonTitle size="large">Team Creation</IonTitle>
+          </IonToolbar>
+        </IonHeader>
+        <IonCard className="create-team-card">
+          <IonCardHeader style={{ display: 'flex', justifyContent: 'center' }}>
+            <img alt="Picture of walktober team" src={create} />

```

Comment:
Could we possibly change this to something like "Art depicting 5 team members jumping in the air"?

Reply:
Done.